src/Octoshift/Services/AwsApi.cs

```diff
@@ -34,9 +35,17 @@ private static AmazonS3Client BuildAmazonS3Client(string awsAccessKeyId, string
             AWSConfigsS3.UseSignatureVersion4 = true;
         }
 
-        return awsSessionToken.HasValue()
-            ? new AmazonS3Client(awsAccessKeyId, awsSecretAccessKey, awsSessionToken, regionEndpoint)
-            : new AmazonS3Client(awsAccessKeyId, awsSecretAccessKey, regionEndpoint);
+        var creds = awsSessionToken.HasValue()
+            ? (AWSCredentials)new SessionAWSCredentials(awsAccessKeyId, awsSecretAccessKey, awsSessionToken)
+            : new BasicAWSCredentials(awsAccessKeyId, awsSecretAccessKey);
+
+        var config = new AmazonS3Config
+        {
+            RegionEndpoint = regionEndpoint,
+            Timeout = TimeSpan.FromMinutes(5)
```

Comment:
Is there any specific thinking behind the 5 minutes number? Should we maybe go for something longer so we can support really large archives? An archive could be up to 10GB, and it's likely that 10GB will take longer than 5 minutes to upload.

```suggestion
            Timeout = TimeSpan.FromMinutes(30)
```

Reply:
We could consider having a configurable timeout, but this seems like a good approach for now.

Reply:
...or is this 5 minutes per part of a multipart upload? If that's the case, then this might be fine?

Reply:
> Is there any specific thinking behind the 5 minutes number?

The default is 100 and I just tripled it.

> ...or is this 5 minutes per part of a multipart upload?

Yes it's for the multi-part uploads 👍

Reply:
The timeout is per part, and the default size of each part is 16MB ([docs](https://docs.aws.amazon.com/mobile/sdkforxamarin/developerguide/s3-integration-transferutility.html#:~:text=value%20is%2010.-,MinSizeBeforePartUpload,-%2D%20Gets%20or%20sets)), so 5 minutes seems like a good number to me. Let's stick with this.